pallets/api/src/nonfungibles/mod.rs

```diff
@@ -0,0 +1,591 @@
+>>::Balance;
+type CollectionIdOf<T> =
+	<NftsOf<T> as Inspect<<T as frame_system::Config>::AccountId>>::CollectionId;
+type ItemIdOf<T> = <NftsOf<T> as Inspect<<T as frame_system::Config>::AccountId>>::ItemId;
+type ItemPriceOf<T> = BalanceOf<T>;
+type CollectionDetailsFor<T> = CollectionDetails<AccountIdOf<T>, BalanceOf<T>>;
+type AttributeNamespaceOf<T> = AttributeNamespace<AccountIdOf<T>>;
+type CollectionConfigFor<T> =
+	CollectionConfig<ItemPriceOf<T>, BlockNumberFor<T>, CollectionIdOf<T>>;
+// Type aliases for pallet-nfts storage items.
+pub(super) type AccountBalanceOf<T> = pallet_nfts::AccountBalance<T, NftsInstanceOf<T>>;
+pub(super) type AttributeOf<T> = pallet_nfts::Attribute<T, NftsInstanceOf<T>>;
+pub(super) type NextCollectionIdOf<T> = pallet_nfts::NextCollectionId<T, NftsInstanceOf<T>>;
+pub(super) type CollectionOf<T> = pallet_nfts::Collection<T, NftsInstanceOf<T>>;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{
+		dispatch::{DispatchResult, DispatchResultWithPostInfo, WithPostDispatchInfo},
+		pallet_prelude::*,
+		traits::Incrementable,
+	};
+	use frame_system::pallet_prelude::*;
+	use pallet_nfts::{CancelAttributesApprovalWitness, DestroyWitness, MintWitness};
+	use sp_runtime::BoundedVec;
+	use sp_std::vec::Vec;
+
+	use super::*;
+
+	/// State reads for the non-fungibles API with required input.
+	#[derive(Encode, Decode, Debug, MaxEncodedLen)]
+	#[cfg_attr(feature = "std", derive(PartialEq, Clone))]
+	#[repr(u8)]
+	#[allow(clippy::unnecessary_cast)]
+	pub enum Read<T: Config> {
+		/// Total item supply of a specified collection.
+		#[codec(index = 0)]
+		TotalSupply(CollectionIdOf<T>),
+		/// Account balance for a specified collection.
+		#[codec(index = 1)]
+		BalanceOf {
+			/// The collection.
+			collection: CollectionIdOf<T>,
+			/// The owner of the collection .
+			owner: AccountIdOf<T>,
+		},
+		/// Allowance for an operator approved by an owner, for a specified collection or item.
+		#[codec(index = 2)]
+		Allowance {
+			/// The collection.
+			collection: CollectionIdOf<T>,
+			/// The collection item.
+			item: Option<ItemIdOf<T>>,
+			/// The owner of the collection item.
+			owner: AccountIdOf<T>,
+			/// The delegated operator of collection item.
+			operator: AccountIdOf<T>,
+		},
+		/// Owner of a specified collection item.
+		#[codec(index = 5)]
+		OwnerOf {
+			/// The collection.
+			collection: CollectionIdOf<T>,
+			/// The collection item.
+			item: ItemIdOf<T>,
+		},
+		/// Attribute value of a specified collection item. (Error: bounded collection is not
+		/// partial)
+		#[codec(index = 6)]
+		GetAttribute {
+			/// The collection.
+			collection: CollectionIdOf<T>,
+			/// The collection item.
+			item: ItemIdOf<T>,
+			/// The namespace of the attribute.
+			namespace: AttributeNamespaceOf<T>,
+			/// The key of the attribute.
+			key: BoundedVec<u8, T::KeyLimit>,
+		},
+		/// Details of a specified collection.
+		#[codec(index = 9)]
+		Collection(CollectionIdOf<T>),
```

Comment:
[NOT MENTIONED IN THE SPEC]

I added this `Collection` read query because I think it is necessary for the smart contract developer to get the details of the NFT collection.